### PR TITLE
lua plugin: add support for relative path scripts; moves inline script to switch

### PR DESCRIPTION
**Pull request by randall**
To use inline scripts, prefix with --inline;
eg:
   @plugin=tslua.so @pparam=--states=4 @pparam=--inline=print(123)

### CODE REVIEW

File: plugins/lua/ts_lua_util.c [hunk 1]
```diff
@@ -201,15 +201,15 @@ ts_lua_add_module(ts_lua_instance_conf *conf, ts_lua_main_ctx *arr, int n, int a
 
     if (conf->content) {
       if (luaL_loadstring(L, conf->content)) {
-        snprintf(errbuf, errbuf_size, "[%s] luaL_loadstring %s failed: %s", __FUNCTION__, conf->script, lua_tostring(L, -1));
+        snprintf(errbuf, errbuf_size, "[%s] luaL_loadstring '%s' failed: %s", __FUNCTION__, conf->content, lua_tostring(L, -1));
```
Comment (shukitchan) [inline plugins/lua/ts_lua_util.c]: i think it may be too much to put the entire script content here. 
Perhaps we can skip that in that ?

**Comment by shukitchan** [comment]
i think it looks good. Just a small comment.

**Comment by randall** [comment]
[approve ci autest]

**Comment by mlibbey** [comment]
Do we need a small doc update with this?

**Comment by randall** [comment]
@mlibbey What do you have in mind here?

**Comment by mlibbey** [comment]
The doc has this example
@pparam=/XXX/test_hdr.lua
Perhaps, "The path referencing a file with the lua script can be relative to the configuration directory or an absolute path."

Then some place in the config section, perhaps
The remap line can contain inline script via the inline keyword. For example
@pparam=--inline=print(123)
would run those lua statements directly.

**Comment by shukitchan** [comment]
I can add that in the weekend if Randall does not beat me to that.

**Comment by zwoop** [comment]
Cherry-picked to 8.1.x

**Pull request merged**